isb_web/sqlmodel_database.py

```diff
@@ -306,6 +361,38 @@ def save_or_update_thing(session: Session, thing: Thing):
                 )
 
 
+def all_thing_identifier_objects(
+    session: Session, min_id: int, batch_size: int
+) -> list[ThingIdentifier]:
+    thing_identifiers_select = (
+        select(ThingIdentifier)
+        .filter(ThingIdentifier.thing_id >= min_id)
+        .order_by(ThingIdentifier.thing_id.asc())
+        .limit(batch_size)
+    )
+    thing_identifiers = session.exec(thing_identifiers_select).all()
+    return thing_identifiers
+
+
+def things_with_null_identifiers(session: Session) -> list[Thing]:
+    # noinspection PyComparisonWithNone
+    things_select = select(Thing).filter(Thing.identifiers == None)
+    things = session.exec(things_select).all()
+    return things
+
+
+def all_thing_identifiers(session: Session, authority: typing.Optional[str] = None) -> typing.Dict[str, int]:
+    thing_identifiers_select = select(Thing.primary_key, Thing.identifiers)
+    if authority is not None:
+        thing_identifiers_select = thing_identifiers_select.where(Thing.authority_id == authority)
+    thing_identifiers = session.execute(thing_identifiers_select).fetchall()
+    thing_identifiers_dict = {}
+    for row in thing_identifiers:
+        for identifier in row[1]:
+            thing_identifiers_dict[identifier] = row[0]
+    return thing_identifiers_dict
```

Comment:
This is the new method for bulk fetching all identifiers and primary keys -- used during import.